README.md

```diff
@@ -662,6 +664,24 @@ type SomeObject = NumberDictionary<{ a: number; b: string }>;
 type EmptyObject = NumberDictionary<{}>;
 ```
 
+### NonEmptyArray
+
+Useful for accepting only arrays containing at least one element.
+
+```typescript
+type NonEmptyRestParameters<T> = NonEmptyArray<T>;
+// Result: [T, ...T[]]
+
+// declare function expression type accepting some rest parameters, but at least one element for the rest parameters is required
+type FunctionAcceptingRestParameters = (someString: string, ...args: NonEmptyRestParameters<number>) => void;
```

Comment:
Not critical comment

You can use it here as is:

```suggestion
type FunctionAcceptingRestParameters = (someString: string, ...args: NonEmptyArray<number>) => void;
```

And then don't forget to remove previous lines

Reply:
Makes sense to simplify the example here a little bit :)